Source/Scene/TimeDynamicPointCloud.js

```diff
@@ -0,0 +1,740 @@
+        this._lastRenderedFrame = undefined;
+        this._clockMultiplier = 0.0;
+
+        // For calculating average load time of the last N frames
+        this._runningSum = 0.0;
+        this._runningLength = 0;
+        this._runningIndex = 0;
+        this._runningSamples = arrayFill(new Array(5), 0.0);
+        this._runningAverage = 0.0;
+    }
+
+    defineProperties(TimeDynamicPointCloud.prototype, {
+        /**
+         * The {@link ClippingPlaneCollection} used to selectively disable rendering the point cloud.
+         *
+         * @memberof TimeDynamicPointCloud.prototype
+         *
+         * @type {ClippingPlaneCollection}
+         */
+        clippingPlanes : {
+            get : function() {
+                return this._clippingPlanes;
+            },
+            set : function(value) {
+                ClippingPlaneCollection.setOwner(value, this, '_clippingPlanes');
+            }
+        },
+
+        /**
+         * The total amount of GPU memory in bytes used by the point cloud.
+         *
+         * @memberof TimeDynamicPointCloud.prototype
+         *
+         * @type {Number}
+         * @readonly
+         *
+         * @see TimeDynamicPointCloud#maximumMemoryUsage
+         */
+        totalMemoryUsageInBytes : {
+            get : function() {
+                return this._totalMemoryUsageInBytes;
+            }
+        },
+
+        /**
+         * The bounding sphere of the frame being rendered. Returns <code>undefined</code> if no frame is being rendered.
+         *
+         * @memberof TimeDynamicPointCloud.prototype
+         *
+         * @type {BoundingSphere}
+         * @readonly
+         */
+        boundingSphere : {
+            get : function() {
+                if (defined(this._lastRenderedFrame)) {
+                    return this._lastRenderedFrame.pointCloud.boundingSphere;
+                }
+            }
+        }
+    });
+
+    function getFragmentShaderLoaded(fs) {
+        return 'uniform vec4 czm_pickColor;\n' + fs;
+    }
+
+    function getUniformMapLoaded(stream) {
+        return function(uniformMap) {
+            return combine(uniformMap, {
+                czm_pickColor : function() {
+                    return stream._pickId.color;
+                }
+            });
+        };
+    }
+
+    function getPickIdLoaded() {
+        return 'czm_pickColor';
+    }
+
+    /**
+     * Marks the point cloud's {@link TimeDynamicPointCloud#style} as dirty, which forces all
+     * points to re-evaluate the style in the next frame.
+     */
+    TimeDynamicPointCloud.prototype.makeStyleDirty = function() {
+        this._styleDirty = true;
+    };
+
+    /**
+     * Exposed for testing.
+     *
+     * @private
+     */
+    TimeDynamicPointCloud.prototype._getAverageLoadTime = function() {
+        if (this._runningLength === 0) {
+            // Before any frames have loaded make a best guess about the average load time
+            return 0.05;
```

Comment:
What's this based on? Can we grab any properties from the browser or system to make a more specific guess?

Reply:
It's basically arbitrary - tuned roughly to a 300kb Draco tile. We would need to know tile size before it's downloaded and network speed to make a proper guess. I don't think it matters too much since the average will get corrected once tiles start to load.